internal/cache/devices_test.go

```diff
@@ -0,0 +1,153 @@
+package cache
+
+import (
+	"context"
+	"testing"
+
+	"github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/mock"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+var ds []models.Device
+
+func init() {
+	common.ServiceName = "device-cache-test"
+	common.DeviceClient = &mock.DeviceClientMock{}
+	ctx := context.WithValue(context.Background(), common.CorrelationHeader, uuid.New().String())
+	ds, _ = common.DeviceClient.DevicesForServiceByName(common.ServiceName, ctx)
+}
+
+func TestNewDeviceCache(t *testing.T) {
+	dc := newDeviceCache([]models.Device{})
+	if _, ok := dc.(DeviceCache); !ok {
+		t.Error("the newDeviceCache function supposed to return a value which holds the DeviceCache type")
+	}
+}
+
+func TestDeviceCache_ForName(t *testing.T) {
+	dc := newDeviceCache(ds)
+
+	if d, found := dc.ForName(mock.ValidDeviceRandomBoolGenerator.Name); !found {
+		t.Error("supposed to find a matching device in cache by a valid device name")
+	} else {
+		assert.Equal(t, mock.ValidDeviceRandomBoolGenerator, d)
+	}
+
+	if _, found := dc.ForName(mock.NewValidDevice.Name); found {
+		t.Error("not supposed to find a matching device in cache by an invalid device name")
+	}
+}
```

Comment:
So here's the ForName implementation
```
func (d *deviceCache) ForName(name string) (models.Device, bool) {
	if device, ok := d.dMap[name]; ok {
		return *device, ok
	} else {
		return models.Device{}, ok
	}
}
```
All this test does is ensure the Go runtime can pull an entry out of a map. That's not a valid unit test.